src/sql/workbench/api/common/extHostModelViewTree.ts

```diff
@@ -161,8 +161,10 @@ export class ExtHostTreeView<T> extends vsTreeExt.ExtHostTreeView<T> {
 	}
 
 	protected createTreeNode(element: T, extensionTreeItem: azdata.TreeComponentItem, parent?: vsTreeExt.TreeNode): vsTreeExt.TreeNode {
-		let item = super.createTreeNode(element, extensionTreeItem, parent);
-		item = Object.assign({}, item, { checked: extensionTreeItem.checked, enabled: extensionTreeItem.enabled });
-		return item;
+		let node = super.createTreeNode(element, extensionTreeItem, parent);
+		if (node.item) {
+			node.item = Object.assign(node.item, { checked: extensionTreeItem.checked, enabled: extensionTreeItem.enabled });
```

Comment:
Should we just cast as an azdata.TreeComponentItem and then directly assign these values?

Reply:
the item is in `ITreeItem` type, and casting to `azdata.TreeComponentItem` fails.
I can directly assign by `node.item['checked'] = extensionTreeItem.checked` though if you want.